src/codecs/gif.rs

```diff
@@ -106,9 +106,29 @@ impl<'a, R: 'a + Read> ImageDecoder<'a> for GifDecoder<R> {
 
         let (width, height) = self.dimensions();
 
-        if (frame.left, frame.top) == (0, 0) && (frame.width, frame.height) != self.dimensions() {
-            // If the frame matches the logical screen, directly read into it
+        if (frame.left, frame.width) == (0, width) && (frame.top as u64 + frame.height as u64 <= height as u64) {
```

Comment:
How about `frame.left == 0 && frame.width == width && ...`?

Reply:
Yes, makes more sense to me too, now that I think of it, given that one is a coordinate and another is a dimension.